pkg/container/kubernetes_runner.go

```diff
@@ -0,0 +1,527 @@
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Command: cmd,
+			Stdout:  true,
+			Stderr:  true,
+		}, scheme.ParameterCodec)
+
+	k.logger.Infof("executing command %v at %s", cmd, req.URL())
+	exec, err := remotecommand.NewSPDYExecutor(k.restConfig, "POST", req.URL())
+	if err != nil {
+		return fmt.Errorf("failed to create remote command executor: %v", err)
+	}
+
+	if err := exec.StreamWithContext(ctx, streamOpts); err != nil {
+		return fmt.Errorf("failed to stream remote command: %v", err)
+	}
+
+	return nil
+}
+
+func (k *k8s) NewBuildPod(ctx context.Context, cfg *Config) (*corev1.Pod, error) {
+	repo, err := name.NewRepository(k.Config.Repo)
+	if err != nil {
+		return nil, err
+	}
+
+	pod := k.Config.defaultBuilderPod(cfg)
+
+	for i, mount := range cfg.Mounts {
+		if k.filterMounts(mount) {
+			continue
+		}
+
+		mountName := fmt.Sprintf("mount-%d", i)
+		bundle, err := kontext.Bundle(ctx, mount.Source, repo.Tag(fmt.Sprintf("%s-%s", cfg.PackageName, mountName)))
+		if err != nil {
+			return nil, err
+		}
+
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, corev1.Container{
+			Name:  mountName,
+			Image: bundle.String(),
+			VolumeMounts: []corev1.VolumeMount{{
+				Name:      mountName,
+				MountPath: mount.Destination,
+			}},
+		})
+		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+			Name: mountName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+		pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      mountName,
+			MountPath: mount.Destination,
+		})
+	}
+
+	return pod, nil
+}
+
+// filterMounts filters mounts that are not supported by the k8s runner
+func (k *k8s) filterMounts(mount BindMount) bool {
+	// the kubelet handles this
+	if mount.Source == "/etc/resolv.conf" {
+		return true
+	}
+
+	// Skip anything that can't be mounted to a destination
+	if mount.Destination == "" {
+		return true
+	}
+
+	// TODO: More filtering?
+	return false
+}
+
+// KubernetesRunnerConfig handles the configuration for the Kubernetes runner
+// It sources from various locations, in the following order of precedence:
+//  1. The "global" config file
+//  2. Defaults
+//
+// TODO: Add loaders from package config and environment
+type KubernetesRunnerConfig struct {
+	Provider    string            `json:"provider" yaml:"provider"`
+	Repo        string            `json:"repo" yaml:"repo"`
+	Namespace   string            `json:"namespace" yaml:"namespace"`
+	Annotations map[string]string `json:"annotations,omitempty" yaml:"annotations,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty" yaml:"labels,omitempty"`
+
+	StartTimeout time.Duration `json:"startTimeout" yaml:"startTimeout"`
+	BuildTimeout time.Duration `json:"buildTimeout" yaml:"buildTimeout"`
+
+	PodTemplate *KubernetesRunnerConfigPodTemplate `json:"podTemplate,omitempty" yaml:"podTemplate,omitempty" ignored:"true"`
```

Comment:
Can we just use the real `corev1.PodTemplate` here?

Reply:
I started that way, then decided it's probably too much rope for end users to strangle themselves with.

granted we're the end user here, so that might be okay?

Reply:
I don't feel strongly either way, and it's not a one-way door. Same with the other comment about `ResourceList`.